src/plugins/Makefile.am

```diff
@@ -291,6 +297,7 @@ AM_LDFLAGS =  $(LDFLAGS)
 AM_LDFLAGS += $(GLIB_LIBS)
 AM_LDFLAGS += $(VMI_LIBS)
 AM_LDFLAGS += $(JSONC_LIBS)
+AM_LDFLAGS += -lstdc++fs
```

Comment:
There should be no manual ldflags added, there should be logic in `configure.ac` that determines if its needed or not. See https://stackoverflow.com/questions/64384496/autotools-stdcfs-false-positive

Reply:
Ok, just read a about how autoconfig, but yet I am not sure what to do.
The code from stackoverflow is 
```
AC_LINK_IFELSE(
    [AC_LANG_SOURCE([
        #if HAVE_FILESYSTEM
        #include <filesystem>
        #endif

        int main() {
            std::error_code ec;
            std::filesystem::create_directory("/dev/null", &ec);
        }
    ])],
    [ac_cv_fs_stdlib=no],
    [ac_cv_fs_stdlib=yes]
)
```

1. As recommended in the reference you mentioned @tklengyel  I put the code block behind the AC_LANG(C++) definition (which is already included as AC_LANG_PUSH(C++) ) 
2. At first I checked for the existence of filesystem header with AC_CHECK_HEADERS which also sets HAVE_FILESYSTEM.
3. Depending on the value of the ac_cv_fs_stdlib variable  I set -lstdc++fs or not

My proposal looks like this:
```
#Check requirement of lstdc++fs
AC_CHECK_HEADER([filesystem])
AC_LINK_IFELSE(
    [AC_LANG_SOURCE([
        #if HAVE_FILESYSTEM
        #include <filesystem>
        #endif

        int main() {
            std::error_code ec;
            std::filesystem::create_directory("/dev/null", &ec);
        }
    ])],
    [ac_cv_fs_stdlib=no],
    [ac_cv_fs_stdlib=yes]
)

if test x$ac_cv_fs_stdlib = xyes; then
    AM_LDFLAGS="$AM_LDFLAGS -lstdc++fs"
    AC_MSG_NOTICE([Added lstdc++fs])
fi
#end requirement check
```

Is it typical that also on Ubuntu 20.04 this lstdc++fs library is used?

Reply:
It crashed badly, but I have no idea how to resolve this.  I guess I need someone's help to resolve this.

Reply:
I have absolutely no idea, why this crashes. I definitely need someone' help. :exploding_head:

Reply:
You need

```
STDCPPFSLIB=""
AC_CHECK_HEADER([filesystem])
AC_LINK_IFELSE(
    [AC_LANG_SOURCE([
        #if HAVE_FILESYSTEM
        #include <filesystem>
        #endif

        int main() {
            std::error_code ec;
            std::filesystem::create_directory("/dev/null", &ec);
        }
    ])], [],
    [STDCPPFSLIB="-lstdc++fs"]
)
AC_SUBST([STCPPFSLIB])
```

and in `src/plugins/Makefile.am`: `AM_LDFLAGS += $(STDCPPFSLIB)`

Reply:
Thanks a lot for the code! I was able to use it. The variable name in AC_SUBST was incorrect (missing 'D') but that was no problem. I changed everything according to your recommendation and it works now. All checks are green